core/app/models/spree/tax/item_adjuster.rb

```diff
@@ -0,0 +1,32 @@
+module Spree
+  module Tax
+    class ItemAdjuster
+      attr_reader :item, :order
+
+      include TaxHelpers
+
+      def initialize(item, options = {})
+        @item = item
+        @order = @item.order
+        # set the instance variable so `TaxRate.match` is only called when necessary
+        @rates_for_order_zone = options[:rates_for_order_zone]
+      end
+
+      def adjust!
```

Comment:
Could we add some YARD here to explain the return values of this method.

I also tend to to not have a method ending in a bang without a corresponding non-bang method.  See http://dablog.rubypal.com/2007/8/15/bang-methods-or-danger-will-rubyist